web/workflow.R

```diff
@@ -39,29 +39,36 @@ if (is.na(args[1])){
   settings <- read.settings(settings.file)
 }
 
+# Update/fix/check settings. Will only run the first time it's called, unless force=TRUE
+settings <- fix.deprecated.settings(settings, force=FALSE)
+settings <- addSecrets(settings, force=FALSE)
+settings <- update.settings(settings, force=FALSE)
+settings <- check.settings(settings, force=FALSE)
+
 # Check for additional modules that will require adding settings
 if("benchmark" %in% names(settings)){
   library(PEcAn.benchmark)
   settings <- papply(settings, read.settings.RR)
 }
 
 # Write pecan.CHECKED.xml
-settings <- write.settings(settings, outputfile = "pecan.CHECKED.xml")
+write.settings(settings, outputfile = "pecan.CHECKED.xml")
```

Comment:
So to build on what @mdietze said: the XML files I'm using (you can see them [here](https://github.com/PecanProject/pecan/tree/master/modules/benchmark/inst/scripts) don't have any of the run information in them. The `read.settings.RR` inserts in everything using the ensemble.id. So the update/fix/addsecrets/check functions need to happen after that section of the workflow is run. Maybe we could combine those into `prepare.settings` and and keep `write.settings` separate. How does that sound?